### ISSUE TITLE
Support shortcut keys on other platforms

### ISSUE THREAD

**Issue by kentcdodds** [kadirahq/storybook-ui#40]
I noticed that the shortcut button is ⌘ in the app and found it [in the code](https://github.com/kadirahq/storybook-ui/blob/2305d1f1109c38a104152fd47a3b681f969dd70c/src/modules/ui/components/left_panel/header.js#L52). It looks like it's hard coded. This makes me wonder whether shortcut keys are supported on other platforms. If so, I think it'd be appropriate to show a different icon in the button that's appropriate for the platform. Thoughts?

**Comment by arunoda** [comment]
Yes. I think we should do it.
May be some other icon which doesn't related to platforms.
Also when we are showing shortcut commands, we should show the correct command for the correct platform.

Eg:- Windows doesn't have a CMD key.

**Comment by m-sureshraj** [comment]
I would like to work on this issue.

Issues:
1. Adding platform independent shortcut icon to the button.
2. Render proper command's according to the platform.

(issue 2) like this: https://support.google.com/chrome/answer/157179?hl=en

![screenshot-from-2016-09-28-23-](https://cloud.githubusercontent.com/assets/12813750/18926395/43563282-85d5-11e6-8cac-95d64c8dbd68.png)

![screenshot-from-2016-09-28-23- 1](https://cloud.githubusercontent.com/assets/12813750/18926457/84ab5384-85d5-11e6-8b72-d8c8a1733616.png)

**Comment by mnmtanish** [comment]
Hi @m-sureshraj 
I think these will be useful: 
 - Shortcut setup [source](https://github.com/kadirahq/storybook-ui/blob/master/src/libs/key_events.js)
 - Shortcuts help component [source](https://github.com/kadirahq/storybook-ui/blob/master/src/modules/ui/components/shortcuts_help.js)

**Comment by m-sureshraj** [comment]
Thank you @mnmtanish 
I will start work on this issue.

**Comment by m-sureshraj** [comment]
Hi @mnmtanish 

I came up this solution to the issue (show the correct command for the correct platform).
I implement the testing solution here - http://jsbin.com/nimazo/edit?js,output

Your feedback will help full. Thank you

**Comment by mnmtanish** [comment]
Awesome :)
Can you open a pull-request with these changes?

You can use the **example** app to preview changes.
https://github.com/kadirahq/storybook-ui/tree/master/example

btw, if you can wait a day or 2 before sending the PR,
you can also sign-up for the Hacktoberfest: [more info](https://hacktoberfest.digitalocean.com/#details)

**Comment by m-sureshraj** [comment]
Thank you @mnmtanish for your feedback :smile:
Actually i already registered for Hacktoberfest, found this issue while searching for Hacktoberfest label.

**Comment by m-sureshraj** [comment]
Hi @mnmtanish 

I finish all the changes. Preview: 
![screenshot-from-2016-10-03-11-](https://cloud.githubusercontent.com/assets/12813750/19028822/fe2de8bc-895d-11e6-8875-1fef2f99a9ec.png)
![screenshot-from-2016-10-03-11- 1](https://cloud.githubusercontent.com/assets/12813750/19028823/fe2e6d6e-895d-11e6-8f17-1782ee1fb87f.png)

**Need Help:**

- What is the equivalent key for (^) in Windows or Linux platform?
- Does this key combination (^ + Shift + P) works other then mac platform?

Thank you.

**Comment by mnmtanish** [comment]
The **^** is used to represent the CTRL key. As only Mac users use that symbol so the shortcut help UI is a bit confusing for Windows/Linux users.

**CTRL+SHIFT+_** shortcut key combinations should work fine on all platforms.

**Comment by m-sureshraj** [comment]
So, it should be better not rendering mac specific key combination to Win/Linux platform.

For Win/Linux platform render: **CTRL + SHIFT + P**
For Mac: **⌘ + SHIFT + P** / **^ + SHIFT + P**

Is that ok?

**Comment by mnmtanish** [comment]
Yeah but use ⇧ for SHIFT in Mac

- Windows/Linux: CTRL SHIFT P
- Mac: ⌘ ⇧ P /  ⌃ ⇧ P

**Comment by m-sureshraj** [comment]
Hi @mnmtanish 

I implement the my solution to this issue. Also opened a PR. Can you please review the PR and give your feedback.

**NOTE** 
BTW i'am new to he **redux** dev. In '**src/modules/ui/containers/shortcuts_help.js**' file i couldn't figure it out, to pass the shortcut data to composer fn properly. I just import the shortcuts fn and invoked inside the composer fn. Actually shortcut fn should be passed as parameter to composer fn like **{ui}, {actions}** I need your guidance on this issue. Thank you.

```
import getShortcuts from '../../../libs/shortcuts';

export const composer = ({ ui }, { actions }) => {
  const actionMap = actions();
  const data = {
    isOpen: ui.showShortcutsHelp,
    onClose: actionMap.ui.toggleShortcutsHelp,
    shortcuts: getShortcuts(),
  };

  return data;
};

export default composeAll(
  reduxComposer(composer),
  useDeps()
)(ShortcutsHelp);
```

**Comment by mnmtanish** [comment]
Fixed with #54

**Comment by mnmtanish** [comment]
Released `v3.6.3`

**Comment by m-sureshraj** [comment]
Hi @mnmtanish 

BTW this is my first PR that get merged. Thank your so much for you guidance on this issue. :+1: 

**Need feedback** 
- Adding platform independent shortcut icon to the button ''. Currently it has ⌘ icon

**Comment by mnmtanish** [comment]
Congratulations! 😄 and Happy Hacktoberfest!

Let's start a new issue for the shortcut icon.
Better discuss this with others before starting
